delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/files/FormDataExtractor.scala

```
@@ -122,12 +127,18 @@ object FormDataExtractor {
             WrappedAkkaRejection(MalformedRequestContentRejection(th.getMessage, th))
         }
 
-      private def extractFile(part: FormData.BodyPart): Future[Option[(FileDescription, BodyPartEntity)]] = part match {
+      private def extractFile(part: FormData.BodyPart): Future[Option[FileInformation]] = part match {
         case part if part.name == fieldName =>
-          val filename    = part.filename.getOrElse("file")
+          val filename = part.filename.getOrElse("file")
+
           val contentType = detectContentType(filename, part.entity.contentType)
+          val metadata    = part.dispositionParams.get("metadata").flatMap { metadata =>
+            println("metadata: " + metadata)
+            parser.parse(metadata).toOption.flatMap(_.as[FileUserMetadata].toOption)
```

We should return an error here